src/libcore/any.rs

```diff
@@ -459,3 +459,28 @@ impl TypeId {
 pub const fn type_name<T: ?Sized>() -> &'static str {
     intrinsics::type_name::<T>()
 }
+
+/// Returns the name of the type of the pointed-to value as a string slice.
+/// This is the same as `type_name::<T>()`, but can be used where the type of a
+/// variable is not easily available.
```

Comment:
> where the type of a variable is not easily available

Could you provide an example of such a case?

Reply:
I had multiple occasions where I wanted to see/explain how the type system works where I ended up using `type_name_of_val` as I don't know of a better way.

- the default type of integers/floats:
```rust
fn type_name_of_val<T>(_val: &T) -> &'static str {
    std::any::type_name::<T>()
}

fn main() {
    let x = 1;
    println!("{}", type_name_of_val(&x));
    let y = 1.0;
    println!("{}", type_name_of_val(&y));
}
```

- checking identifier resolution:
```rust
fn type_name_of_val<T>(_val: &T) -> &'static str {
    std::any::type_name::<T>()
}

mod inner {
    pub struct A;
}

fn main() {
    let A = 7;
    {
        use inner::*;
        println!("{}", type_name_of_val(&A));
    }
}
```

Reply:
> ```rust
> fn main() {
>     let x = 1;
>     println!("{}", type_name_of_val(&x));
>     let y = 1.0;
>     println!("{}", type_name_of_val(&y));
> }
> ```

Can you add this as an example to the docs?